File: app/src/main/java/com/fionera/demo/activity/SmartTabLayoutActivity.java
package com.fionera.demo.activity;

import android.os.Bundle;

import com.fionera.base.util.ShowToast;
import com.fionera.demo.R;
import com.fionera.demo.fragment.ExtrasFragment;
import com.ogaclejapan.smarttablayout.SmartTabLayout;

import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

/**
 * @author fionera
 */
public class SmartTabLayoutActivity
        extends AppCompatActivity {

    private String[] title = {"你好", "我并不好", "我真的很不好", "我并不好", "我真的很不好"};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_smart_tab_layout);

        Toolbar toolbar = findViewById(R.id.toolbar);
        toolbar.inflateMenu(R.menu.menu_test);
        toolbar.setOnMenuItemClickListener(item -> {
            ShowToast.show(item.getTitle());
            return true;
        });

        DrawerLayout drawer = findViewById(R.id.drawer);

        ActionBarDrawerToggle drawerToggle = new ActionBarDrawerToggle(this, drawer, toolbar, 0, 0);
        drawerToggle.syncState();
        drawer.addDrawerListener(drawerToggle);

        ViewPager viewPager = findViewById(R.id.vp_tab_layout);
        viewPager.setAdapter(new FragmentPagerAdapter(getSupportFragmentManager()) {
            @Override
            public Fragment getItem(int position) {
                return new ExtrasFragment();
            }

            @Override
            public int getCount() {
                return title.length;
            }

            @Override
            public CharSequence getPageTitle(int position) {
                return title[position % title.length];
            }
        });

        SmartTabLayout tabLayout = findViewById(R.id.tl_tab_layout);
        tabLayout.setViewPager(viewPager);
    }
}

File: app/src/main/java/com/fionera/demo/activity/SmartTabLayoutActivity.kt
package com.fionera.demo.activity

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentPagerAdapter
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import com.fionera.demo.R
import com.fionera.demo.fragment.TabLayoutFragment
import com.fionera.demo.util.ShowToast
import kotlinx.android.synthetic.main.activity_smart_tab_layout.*
import org.jetbrains.anko.find

class SmartTabLayoutActivity : AppCompatActivity() {

    private val title = arrayOf("你好", "我并不好", "我真的很不好", "我并不好", "我真的很不好")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_smart_tab_layout)

        val toolbar = find<Toolbar>(R.id.toolbar)
        toolbar.inflateMenu(R.menu.menu_recycle)
        toolbar.setOnMenuItemClickListener { item ->
            ShowToast.show(item.title)
            true
        }

        val drawerToggle = ActionBarDrawerToggle(this, drawer, toolbar, 0, 0)
        drawerToggle.syncState()
        drawer.addDrawerListener(drawerToggle)

        vp_tab_layout!!.adapter = object : FragmentPagerAdapter(supportFragmentManager) {
            override fun getItem(position: Int): Fragment {
                return TabLayoutFragment()
            }

            override fun getCount(): Int {
                return title.size
            }

            override fun getPageTitle(position: Int): CharSequence {
                return title[position % title.size]
            }
        }

        tl_tab_layout!!.setViewPager(vp_tab_layout)
    }
}
